great_expectations/data_context/types/base.py

```diff
@@ -147,6 +147,10 @@ def __init__(
         max_keys=None,
         schema_name=None,
         batch_spec_passthrough=None,
+        group_names: Optional[List[str]] = None,
+        pattern: Optional[str] = None,
+        base_directory: Optional[List[str]] = None,
+        glob_directive: Optional[str] = None,
```

Comment:
These are noted in the corresponding `Schema` class but not the config. I ran the entire test suite and noted that these were the 4 args that were dynamically being set through `setattr`.

Explicitly noting them here tightens up our schema/config and allows us to get rid of the dynamic `setattr` behavior.